packages/file_selector/file_selector_windows/example/lib/save_text_page.dart

```diff
@@ -17,19 +18,37 @@ class SaveTextPage extends StatelessWidget {
 
   Future<void> _saveFile() async {
     final String fileName = _nameController.text;
-    final String? path = await FileSelectorPlatform.instance.getSavePath(
-      // Operation was canceled by the user.
-      suggestedName: fileName,
+    final FileSaveLocationResult? result =
+        await FileSelectorPlatform.instance.getSaveLocation(
+      options: SaveDialogOptions(suggestedName: fileName),
+      acceptedTypeGroups: const <XTypeGroup>[
+        XTypeGroup(
+          label: 'Plain text',
+          extensions: <String>['txt'],
+        ),
+        XTypeGroup(
+          label: 'JSON',
+          extensions: <String>['json'],
+        ),
+      ],
     );
-    if (path == null) {
+    // Operation was canceled by the user.
+    if (result == null) {
       return;
     }
+    String path = result.path;
+    // Append an extension based on the selected type group if the user didn't
+    // include one.
+    if (!path.split(Platform.pathSeparator).last.contains('.')) {
+      final XTypeGroup? activeGroup = result.activeFilter;
+      if (activeGroup != null) {
+        path = '$path.${activeGroup.extensions!.first}';
```

Comment:
Is the extensions list always guaranteed to contain at least one so long as there's an activeFilter? (That makes logical sense but it's a bummer we need to know it here)

Reply:
This is dependent on the code 15-20 lines up where we define the groups. I'll add a comment explaining that so it's clear to a reader why it's safe.